Remove FindUBUsingMinMaxD algorithm

sf1919 opened the pull request:
### Description of work
Removal of FindUBUsingMinMaxD Algorithm as this has been deprecated in 2013.

#### Summary of work
Remove FindUBUsingMinMaxDs, FindUBUsingMinMaxDTest and associated files from Mantid Code base.

This PR also removes references from CMake and documentation references from v6.1.0 release notes for diffraction.

Part of #31258


#### Further detail of work
Deprecated as part of commit [#b65b56e](https://github.com/mantidproject/mantid/commit/b65b56e09482dbdbf399637a8cff7f7510f48ec6)

### To test:

- tests should pass
- Check correct files have been removed (algorithm & tests)
- Check correct lines have been removed from CMake
- Check there are no further links/references to FindUBUsingMinMaxD in the code base



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

sf1919 committed:
Remove FindUBUsingMinMaxD algorithm

- Remove algorithm inc and src files
- Update Cmake list

sf1919 committed:
Remove tests for FindUBUsingMinMaxD

- Remove test file
- Update CMakeLists

sf1919 committed:
Update documentation for FindUBUsingMinMaxD

- Remove documentation for algorithm
- Remove link from v6.1.0 Diffraction release notes

sf1919 committed:
Add release note

RichardWaiteSTFC reviewed (changes requested):
Thanks for this, I think there are still some references to `FindUBUsingMinMaxD` in related algorithms that should be removed
https://github.com/mantidproject/mantid/blob/a3f64a421d20b9614f5b5a8578aaca01da7da334/Framework/Crystal/inc/MantidCrystal/FindUBUsingFFT.h#L28
https://github.com/mantidproject/mantid/blob/a3f64a421d20b9614f5b5a8578aaca01da7da334/Framework/Crystal/inc/MantidCrystal/FindUBUsingIndexedPeaks.h#L30
https://github.com/mantidproject/mantid/blob/a3f64a421d20b9614f5b5a8578aaca01da7da334/Framework/Crystal/inc/MantidCrystal/FindUBUsingLatticeParameters.h#L28
https://github.com/mantidproject/mantid/blob/a3f64a421d20b9614f5b5a8578aaca01da7da334/Framework/Crystal/inc/MantidCrystal/SetUB.h#L34

sf1919 committed:
Remove see also references to FindUBUsingMinMaxD

- Removed from FindUBUsingFFT documentation
- Removed from FindUBUsingIndexedPeaks documentation
- Removed from FindUBUsingLatticeParameters documentation
- Removed from SetUB documentation

RichardWaiteSTFC reviewed (approved):
Looks good and CI passes so approving!
